eox_tenant/management/commands/create_or_update_tenant_config.py

```diff
@@ -0,0 +1,122 @@
+"""
+Create or updates the TenantConfig for given routes.
+"""
+
+import codecs
+import json
+import logging
+
+from django.core.management import BaseCommand
+from jsonfield.fields import JSONField
+import six
+
+from eox_tenant.models import Route, TenantConfig
+
+LOG = logging.getLogger(__name__)
+
+
+def load_json_from_file(filename):
+    """
+    Loads json content from file.
+    """
+    with codecs.open(filename, encoding='utf-8') as file:
+        return json.load(file)
+
+
+class Command(BaseCommand):
+    """
+    Management command to create or update TenantConfig.
+    """
+    help = 'Create or update TenantConfig'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--external-key',
+            action='store',
+            dest='external_key',
+            required=True,
+            type=str,
+            help='External key of the tenant config'
+        )
+        parser.add_argument('routes', nargs='+', help='Routes to link to this tenant config')
+
+        parser.add_argument(
+            '--config',
+            type=json.loads,
+            help="Enter JSON tenant configurations",
+            required=False
+        )
+        parser.add_argument(
+            '-f',
+            '--config-file',
+            type=load_json_from_file,
+            dest='config_file_data',
+            help="Enter the path to the JSON file containing the tenant configuration",
+            required=False
+        )
+
+    def merge_dict(self, base_dict: dict, override: dict):
+        """
+        Merge two nested dicts.
+        """
+        for key, value in override.items():
+            if isinstance(value, dict):
+                merged = base_dict.get(key, {}).copy()
+                merged.update(value)
+                base_dict[key] = merged
+                continue
+            base_dict[key] = value
+        return base_dict
```

Comment:
This line
```
merged = base_dict.get(key, {}).copy()
```
throws an error if the value in `base_dict` is not a dict itself. This is possible in scenarios where some config key has a string configured, but we later decide to change it to a dict.

We can perhaps do this recursively, something like this:
```suggestion
    def merge_dict(self, base_dict: dict, override: dict):
        """
        Merge two nested dicts.
        """
        if isinstance(base_dict, dict) and isinstance(override, dict):
            for key, value in override.items():
                base_dict[key] = self.merge_dict(base_dict.get(key, {}), value)
            return base_dict
        
        return override
```

Reply:
@kaustavb12 This condition cannot happen here as `base_dict` will always be a dict since it is fetched from db after checking that the field is of type `JSONField` on line 104:
```python
            for field in TenantConfig._meta.get_fields():
                if isinstance(field, JSONField):
                    name = field.name
                    value = tenant_configuration_values.get(name)
                    if value:
                        base_value = getattr(tenant, name, {})
                        merged = self.merge_dict(base_value, value)

```

Reply:
@navinkarkera  Sorry, I wasn't clear about this earlier. What I meant here was, if the value for a particular key in `base_dict` in not a dict itself, then this throws an error.

For example, if the old config is :
```
{
    "lms_configs": {
        "KEY1": "VALUE1"
    }
}
```
and the new config is 
```
{
    "lms_configs": {
        "KEY1": {
            "INNER_KEY1": "VALUE1"
        }
    }
}
```
In this case, since the value for `KEY1` in the new config is a dict, we try to create a copy of the old value in this line
```
merged = base_dict.get(key, {}).copy()
```
This throws `AttributeError: 'str' object has no attribute 'copy'`

Reply:
@kaustavb12 Ahh yes, that makes sense. Thanks!